.github/workflows/build-latex.yml

```diff
@@ -0,0 +1,38 @@
+name: Build LaTeX document
+on: [push]
+jobs:
+  build_latex:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Git repository
+        uses: actions/checkout@v1
+      - name: Compile LaTeX document
+        uses: xu-cheng/latex-action@master
+        with:
+          root_file: wp.tex
+      - name: Set env
+        if: github.ref == 'refs/heads/master'
+        run: echo ::set-env name=GITHUB_SHA_SHORT::$(echo $GITHUB_SHA | cut -c 1-6)
+      - name: Create Release
+        if: github.ref == 'refs/heads/master'
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        with:
+          tag_name: ${{ env.GITHUB_SHA_SHORT }}
+          release_name: Release ${{ env.GITHUB_SHA_SHORT }}
+          body: Auto-generated release
+          draft: false
+          prerelease: false
+      - name: Upload Release Asset
+        if: github.ref == 'refs/heads/master'
```

Comment:
@Sammers21 instead of this condition you can use event syntax:
```yaml
on:
  push:
    branches:
      - master
```
instead of
```yaml
on: [push]
```
see https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#example-using-multiple-events-with-activity-types-or-configuration